### DISCUSSION TITLE
OPDS catalog view should inform wich book is already downloaded

### DISCUSSION THREAD

**Discussion by gautierchomel** [edrlab/thorium-reader#1863] [Developer corner]
While browsing an OPDS catalog we have no information about books from the feed already downloaded and present in local catalog. 

If importing a file already in the local catalog, download process happens (download bar appears) then we get a notification _FileName was already imported_

A basic information should be given about books already downloaded.

**Comment by danielweck** [comment]
Yes, it is very frustrating not being able to cross-reference from the local bookshelf to the originating OPDS feed (if any), and vice-versa.

**Comment by panaC** [comment]
Technically how to do this ? 

There are no information in an OPDS catalog to formally identify a publication.


From the OPDSPublication element [spec](https://drafts.opds.io/opds-2.0#12-publications):
- we have an identifier key which can contains a uniq uuid, the isbn or whatever. it depends on the implementation of opds server.

- we have the download link. This link is uniq to download the resources but not bijective. The link may contains a random signing token in it. We cannot rely on this link, it depends too much on the implementation server.

- the book name with the server url. A concatenation of the book name with the server url to identify the book from the server can be the solution. A cross-reference between 2 opds server cannot be possible, but from a naive solution. It allows to indicate that this book has already be downloaded from this server, but doesn't certify that is the same book if the opds server updates its collection of books.

- an another solution is to add a hash to the ressource in the opds publication metadata. Currently it's not from the spec and not implemented on any opds server. This hash is like an md5 sha1 or sha256 allow to be sure that the publication is formally the same as on the server.
